tests/pipelines/main.nf.test

```diff
@@ -1102,4 +1104,109 @@ nextflow_pipeline {
             assert iridanext_metadata.isEmpty()
         }
     }
+
+    test("Test zipped output"){
+        tag "zip-output"
+        when {
+            params {
+                input = "$baseDir/tests/data/samplesheets/samplesheet.csv"
+                outdir = "results"
+
+                zip_cluster_results = true
+
+                metadata_partition_name = "outbreak"
+                metadata_1_header = "organism"
+                metadata_2_header = "subtype"
+                metadata_3_header = "country"
+                metadata_4_header = "serovar"
+                metadata_5_header = "age"
+                metadata_6_header = "date"
+                metadata_7_header = "source"
+                metadata_8_header = "special"
+            }
+        }
+
+        then {
+            assert workflow.success
+            assert path("$launchDir/results").exists()
+
+            // Unzip the file for testing
+            def zipfile = new ZipFile("$launchDir/results/zip/arboratornf.zip")
```

Comment:
There's actually a plugin in `nf-test` for working with zip files: https://github.com/lukfor/nft-compress. You could look into using that instead since I think it would simplify a bit of code.